src/CommandLineParser.cc

```diff
@@ -209,25 +209,60 @@ void CommandLineParser::parse(std::span<char*> argv)
 			break;
 		case PHASE_DEFAULT_MACHINE: {
 			if (!haveConfig) {
-				// load default config file in case the user didn't specify one
-				const auto& machine =
-					reactor.getDefaultMachineSetting().getString();
-				try {
-					reactor.switchMachine(string(machine));
-				} catch (MSXException& e) {
-					reactor.getCliComm().printInfo(
-						"Failed to initialize default machine: ",
-						e.getMessage());
-					// Default machine is broken; fall back to C-BIOS config.
-					auto fallbackMachine = std::string(
-						reactor.getDefaultMachineSetting().getDefaultValue().getString());
-					reactor.getCliComm().printInfo(
-						"Using fallback machine: ", fallbackMachine);
+				// load default setup in case the user didn't specify one
+				const auto& defaultSetup =
+					reactor.getDefaultSetupSetting().getString();
+				if (!defaultSetup.empty()) {
+					auto context = userDataFileContext(Reactor::SETUP_DIR);
+					std::string filename;
 					try {
-						reactor.switchMachine(fallbackMachine);
-					} catch (MSXException& e2) {
-						// Fallback machine failed as well; we're out of options.
-						throw FatalError(std::move(e2).getMessage());
+						// Try filename as typed by user in setting
+						filename = context.resolve(defaultSetup);
+					} catch (MSXException&) {
+						// Not found, try adding the normal extension
+						try {
+							filename = context.resolve(tmpStrCat(
+								defaultSetup, Reactor::SETUP_EXTENSION));
```

Comment:
A general observation, not directly related to the code itself:
* For machines we use the `.xml` extension. But we never expose that extension to the user. Instead the user always uses the machine name without that extension.

So maybe we should do the same for setups?
* On save always, unconditionally, add the `.oms` extension.
* And same on load, always look for the filename with an added `.oms` extension.

This will simplify the code, we don't need to try each load twice. And it will also hide the internal implementation detail better that setups are in fact implemented as savestates.

Reply:
I think you're right and I changed the code to do this.

(Not sure that all code we have for other files already does this, but that's for later...)